test/test_windows.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+
+from . import test_projects, utils
+
+basic_project = test_projects.new_c_project()
```

Comment:
This does not include a `pyproject.toml`! I'm almost sure adding one will break this PR.

Reply:
If you're talking about pip's build isolation, pip used to enable out-of-tree builds when a pyproject.toml was present, but that feature was turned off if I remember correctly. https://github.com/pypa/pip/issues/7555

Reply:
It's not out of tree vs. in tree, it's the fact I think this patch hacks the local setuptools instead of the venv setuptools. Both `build` and pip create a fresh environment for building if pyproject.toml is present. The main difference is I think `build` always does, even if the toml file missing. There has been talk about making pip do this too, and pip is considering switching to use `build` to do the building.

Personally would be okay if the environment we create was complexity barren and didn't have setuptools, encouraging use of isolated builds and pyproject.toml. :) But probably too drastic.